hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestReconAsPassiveScm.java

```diff
@@ -0,0 +1,189 @@
+        scmContainerManager.allocateContainer(RATIS, ONE, "test");
+    long containerID = containerInfo.getContainerID();
+    Pipeline pipeline =
+        scmPipelineManager.getPipeline(containerInfo.getPipelineID());
+    XceiverClientGrpc client = new XceiverClientGrpc(pipeline, conf);
+    runTestOzoneContainerViaDataNode(containerID, client);
+
+    // Verify Recon picked up the new container that was created.
+    assertEquals(scmContainerManager.getContainerIDs(),
+        reconContainerManager.getContainerIDs());
+  }
+
+  @Test(timeout = 120000)
+  public void testReconRestart() throws Exception {
+    final OzoneStorageContainerManager reconScm =
+            cluster.getReconServer().getReconStorageContainerManager();
+    StorageContainerManager scm = cluster.getStorageContainerManager();
+
+    // Stop Recon
+    ContainerManager scmContainerManager = scm.getContainerManager();
+    assertTrue(scmContainerManager.getContainerIDs().isEmpty());
+    ContainerManager reconContainerManager = reconScm.getContainerManager();
+    assertTrue(reconContainerManager.getContainerIDs().isEmpty());
+
+    LambdaTestUtils.await(60000, 5000,
+        () -> (reconScm.getScmNodeManager().getAllNodes().size() == 3));
+
+    cluster.stopRecon();
+
+    // Create container in SCM.
+    ContainerInfo containerInfo =
+        scmContainerManager.allocateContainer(RATIS, ONE, "test");
+    long containerID = containerInfo.getContainerID();
+    PipelineManager scmPipelineManager = scm.getPipelineManager();
+    Pipeline pipeline =
+        scmPipelineManager.getPipeline(containerInfo.getPipelineID());
+    XceiverClientGrpc client = new XceiverClientGrpc(pipeline, conf);
+    runTestOzoneContainerViaDataNode(containerID, client);
+    assertFalse(scmContainerManager.getContainerIDs().isEmpty());
+
+    // Close a pipeline
+    Optional<Pipeline> pipelineToClose = scmPipelineManager
+        .getPipelines(RATIS, ONE)
+        .stream()
+        .filter(p -> !p.getId().equals(containerInfo.getPipelineID()))
+        .findFirst();
+    assertTrue(pipelineToClose.isPresent());
+    scmPipelineManager.finalizeAndDestroyPipeline(pipelineToClose.get(), false);
+
+    // Start Recon
+    cluster.startRecon();
+
+    // Verify if Recon has all the nodes on restart (even if heartbeats are
+    // not yet received).
+    NodeManager reconNodeManager = reconScm.getScmNodeManager();
+    NodeManager scmNodeManager = scm.getScmNodeManager();
+    assertEquals(scmNodeManager.getAllNodes().size(),
+        reconNodeManager.getAllNodes().size());
+
+    // Verify Recon picks up new container, close pipeline SCM actions.
+    OzoneStorageContainerManager newReconScm =
+        cluster.getReconServer().getReconStorageContainerManager();
+    PipelineManager reconPipelineManager = newReconScm.getPipelineManager();
+    assertFalse(
+        reconPipelineManager.containsPipeline(pipelineToClose.get().getId()));
+
+    LambdaTestUtils.await(300000, 30000,
```

Comment:
Can we reduce the interval from 30s to 5s?

Reply:
Yes, will do.